source/Mission.cpp

```diff
@@ -808,7 +809,8 @@ bool Mission::Do(Trigger trigger, PlayerInfo &player, UI *ui, const shared_ptr<S
 	}
 	// Don't update any conditions if this action exists and can't be completed.
 	auto it = actions.find(trigger);
-	if(it != actions.end() && !it->second.CanBeDone(player, boardingShip))
+	bool hasAction = it != actions.end();
+	if(hasAction && !it->second.CanBeDone(player, boardingShip))
```

Comment:
I would prefer to handle the alternative action lookup here, because I'm afraid that you could get double counts for aborted if `return Do(FAIL, player, ui, boardingShip);` on line 870 returns false for some reason when the `++player.Conditions()[name + ": aborted"];` on line 846 was already executed.

(Note that this suggestion is incomplete, the hasAction boolean also needs to be updated after the second find or needs to be removed and reverted to the original code of comparing `it` with `actions.end()`.)

```suggestion
	if(it == actions.end() && trigger == ABORT)
		it = actions.find(FAIL)
	if(it != actions.end() && !it->second.CanBeDone(player, boardingShip))
```

Reply:
Good catch.